pkg/syncer/syncer.go

```diff
@@ -326,21 +328,18 @@ func (c *Controller) handleErr(err error, i interface{}) {
 	klog.Errorf("Dropping key %q after failed retries: %v", i, err)
 }
 
-const nsDelimiter = "--"
-const nsTemplate = "kcp" + nsDelimiter + "%s" + nsDelimiter + "%s"
-
-func toCluster(clusterName, kcpNamespace string) (namespace string) {
-	return fmt.Sprintf(nsTemplate, clusterName, kcpNamespace)
+type namespaceLocator struct {
+	LogicalCluster string `json:"logical-cluster"`
+	Namespace      string `json:"namespace"`
 }
 
-func toKcp(namespace string) (clusterName, kcpNamespace string, err error) {
-	segments := strings.Split(namespace, nsDelimiter)
-
-	if len(segments) != 3 {
-		return "", "", fmt.Errorf("Failed to transform name: %s", namespace)
+func physicalClusterNamespaceName(l namespaceLocator) (string, error) {
+	b, err := json.Marshal(l)
+	if err != nil {
+		return "", err
 	}
-
-	return segments[1], segments[2], nil
+	hash := sha256.Sum224(b)
+	return fmt.Sprintf("kcp%x", hash), nil
```

Comment:
Maybe comment why we do this prefixing?

Reply:
I can remove unless you think it's useful?

Reply:
I think we want to retain this prefix so the provenance of a given namespace is obvious from its name (i.e. visible with `kubectl get <type>`) vs having to inspect individual objects.

In keeping with human-friendly UX, maybe insert a separator for readability (e.g. `kcp-<hash>`)?